util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/rule/TestElasticsearchClient.java

```diff
@@ -486,13 +489,26 @@ private JsonObject buildStructuredSettings(String settingsPath, String settings)
 			for ( String property : components ) {
 				JsonObject parent = new JsonObject();
 				parent.add( property, settingsJsonElement );
+
+				if ( withMaxResultWindow && "index".equals( property )
+						&& settingsJsonElement instanceof JsonObject ) {
+					addDefaultMaxResultWindow( (JsonObject) settingsJsonElement );
+				}
+
 				settingsJsonElement = parent;
 			}
 		}
 
 		return settingsJsonElement.getAsJsonObject();
 	}
 
+	private void addDefaultMaxResultWindow(JsonObject settingsJsonElement) {
```

Comment:
Again, we shouldn't have validation errors when the max result window is the default. The validation code should assume that `null` means `10,000`. I left a comment on the validation code with a solution.